orchagent/intfsorch.cpp

```diff
@@ -250,26 +235,74 @@ bool IntfsOrch::addRouterIntfs(Port &port, sai_object_id_t virtual_router_id, Ma
         return false;
     }
 
-    m_portsOrch->setPort(port.m_alias, port);
+    gPortsOrch->setPort(port.m_alias, port);
 
     SWSS_LOG_NOTICE("Create router interface for port %s", port.m_alias.c_str());
 
     return true;
 }
 
-bool IntfsOrch::removeRouterIntfs(Port &port)
+bool IntfsOrch::removeRouterIntfs(string alias, IpPrefix ip_prefix)
 {
     SWSS_LOG_ENTER();
 
-    sai_status_t status = sai_router_intfs_api->remove_router_interface(port.m_rif_id);
+    if (m_syncdIntfses.find(alias) == m_syncdIntfses.end())
+        return true;
+
+    if (m_syncdIntfses[alias].ref_count > 2)
```

Comment:
>if (m_syncdIntfses[alias].ref_count > 2) [](start = 4, length = 40)

for subnet and interface address IP route, can we count them as internal ref_count?